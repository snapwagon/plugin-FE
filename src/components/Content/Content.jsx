import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

const Content = (props) => {
  const subtitleClasses = {
    'snapW-Content__subtitle--is-bold': props.tagline
  };

  return (
    <div className="snapW-Content">
      <p className="snapW-Content__text snapW-Content__title">{props.title}</p>
      {props.subtitle && (
        <p className={cx('snapW-Content__text', 'snapW-Content__subtitle', subtitleClasses)}>
          {props.subtitle}
        </p>
      )}
      {props.tagline && <p className="snapW-Content__text snapW-Content__tagline">{props.tagline}</p>}
    </div>
  );
};

const {
  string
} = PropTypes;

Content.propTypes = {
  subtitle: string,
  tagline: string,
  title: string.isRequired,
  details: string
};

Content.defaultProps = {
  subtitle: '',
  tagline: '',
};

export default Content;
