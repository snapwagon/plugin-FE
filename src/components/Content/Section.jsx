import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

const Section = (props) => {
  return (
    <div
      className={cx(
        `coup-Card__${props.type}`,
        `coup-Card__${props.type}--${props.orientation}`,
        props.classNames
      )}
    >
      {props.children}
    </div>
  );
};

const {
  arrayOf,
  bool,
  node,
  objectOf,
  oneOfType,
  oneOf,
  string,
} = PropTypes;

Section.propTypes = {
  children: oneOfType([
    arrayOf(node),
    node
  ]).isRequired,
  classNames: oneOfType([
    string,
    objectOf(bool)
  ]),
  orientation: oneOf([
    'landscape',
    'portrait'
  ]),
  type: oneOf([
    'Header',
    'Body',
    'Footer'
  ]).isRequired
};

Section.defaultProps = {
  classNames: undefined,
  orientation: 'portrait'
};

export default Section;
