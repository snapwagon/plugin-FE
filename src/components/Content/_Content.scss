.coup-Content {
  box-sizing: border-box;
  padding: 16px 0 0 16px;
  margin: 0;
  font-size: $medium;
  overflow: hidden;
  max-width: 277px;
}

.coup-Content__text {
  display: block;
  margin: 0;
  padding: 0;
  line-height: 20px;
  text-overflow: ellipsis;
}

.coup-Content__title {
  font-size: 1em;
  font-weight: bold;
  color: $blue;
  font-family: $lato;
  line-height: 22px;
}

.coup-Content__subtitle {
  font-family: $lato;
  font-size: 0.875em;
  color: $gray;
}

.coup-Content__subtitle--is-bold {
  font-weight: $med;
}

.coup-Content__tagline {
  font-size: 0.875em;
  color: $gray;
}

.coup-Card--is-bordered {
  box-shadow: 0 1px 4px 0 rgba(0,0,0,0.08);
  transition: box-shadow 0.3s ease-in-out;
  border: 1px solid #E4E6EB;

  &:hover {
    box-shadow: 0 1px 4px 0 rgba(0,0,0,0.2), inset 0 0 0 1px rgba(240,242,247,1);
  }
}

.coup-Card {
  position: relative;
  box-sizing: border-box;
  display: flex;
  border-radius: 4px;
  box-shadow: 0 1px 4px 0 rgba(0,0,0,0.08);
  transition: box-shadow 0.3s ease-in-out;
  font-family: $lato;
  background-color: $white;

  &--landscape {
    flex-direction: row;
    flex-wrap: wrap;
  }

  &--portrait {
    flex-direction: column;
    border: 1px solid #E4E6EB;
    margin: 2px;
  }

  &--small {
    width: 220px;
  }

  &--medium {
    width: 283px;
  }

  &--large {
    width: 350px;
  }

  // Gross
  &--minimum-height {
    &.coup-Card--portrait {
      min-height: 362px;
    }

    .coup-Cards--row-scroll &.coup-Card--portrait {
      min-height: 214px;
    }

    &.coup-Card--landscape {
      min-height: 168px;
    }
  }

  &--is-disabled {
    @include disabled;
  }
}

.coup-Card__Header {
  display: flex;
  flex-direction: column;
  position: relative;
  overflow: hidden;

  &--landscape {
    flex: 1;
  }

  &--portrait {
    flex: 1;
    min-height: 150px;

    .coup-Cards--row-scroll & {
      flex: 3;
      margin-bottom: 10px;
    }
  }

  &--landscape {
    flex: 1;
  }

}

.coup-Card__Body {
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  &--landscape {
    flex: 2;
    overflow: hidden;
  }

  &--portrait {
    flex: 1;
    min-height: 100px;

    .coup-Cards--row-scroll & {
      flex: 2;
      margin-bottom: 10px;
    }
  }
}

.coup-Card__Footer {
  margin-top: auto;

  &--landscape {
    flex: 1 0 100%;
  }

  &--portrait {
    width: 100%;
  }
}

.coup-Section--split {
  flex: 1 0 50%;
  height: 120px;
}

.coup-Card--is-selected .coup-Content--action {
  transform: translateY(0px);
  text-decoration: none;
}

.content-slide-up-enter {
  transform: translateY(100%);
  background-color: $lightBlueGray;

  &.content-slide-up-enter-active {
    transform: translateY(0);
    transition: transform 0.6s ease-in-out;
  }
}

.content-slide-up-leave {
  transform: translateY(0);

  &.content-slide-up-leave-active {
    transform: translateY(-100%);
    transition: transform 0.6s ease-in-out;
    background-color: $lightBlueGray;
  }
}
