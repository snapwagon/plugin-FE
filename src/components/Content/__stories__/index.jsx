/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import { storiesOf } from '@kadira/storybook';

import Content from '../Content.jsx';

storiesOf('Content', module)
  .addWithInfo('rendered with title and subtitle', () => (
    <Content
      subtitle="with Pineapple & Pickled JalapeÃ±o Salsa"
      title="Pork & Cabbage Tacos"
    />
  ))
  .addWithInfo('rendered with title, subtitle and tagline', () => (
    <Content
      title="Riesling"
      subtitle="Le P'tit Paysan"
      tagline="California, 2016"
    />
  ));
