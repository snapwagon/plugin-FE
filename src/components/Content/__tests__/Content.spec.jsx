/* eslint-disable no-undef */

import React from 'react';
import renderer from 'react-test-renderer';

import Content from '../Content.jsx';

describe('<Content /> snapshots', () => {
  test('rendered with title, subtitle and tagline', () => {
    const tree = renderer.create(
      <Content
        title="Food"
        subtitle="Tastes goood"
        tagline="yupp"
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('rendered with title and subtitle', () => {
    const tree = renderer.create(
      <Content
        title="Food"
        subtitle="Tastes goood"
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

