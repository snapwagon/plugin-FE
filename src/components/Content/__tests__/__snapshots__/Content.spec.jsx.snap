// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> snapshots rendered with title and subtitle 1`] = `
<div
  className="snapW-Content"
>
  <p
    className="snapW-Content__text snapW-Content__title"
  >
    Food
  </p>
  <p
    className="snapW-Content__text snapW-Content__subtitle"
  >
    Tastes goood
  </p>
</div>
`;

exports[`<Content /> snapshots rendered with title, subtitle and tagline 1`] = `
<div
  className="snapW-Content"
>
  <p
    className="snapW-Content__text snapW-Content__title"
  >
    Food
  </p>
  <p
    className="snapW-Content__text snapW-Content__subtitle snapW-Content__subtitle--is-bold"
  >
    Tastes goood
  </p>
  <p
    className="snapW-Content__text snapW-Content__tagline"
  >
    yupp
  </p>
</div>
`;
